tests/Session/test_session.py

```diff
@@ -1,14 +1,92 @@
+
     @patch('src.Session.session.input', create=True)
     def testStart(self, mocked_input):
-        # TODO: This can't be tested right now until we do regression tests
-        pass
\ No newline at end of file
+        # This skips checking the welcome message, since twoPrintOutcomes
+        # has been tested in utils and it depends on someone's personal profiles.
+
+        # Testing that the 'exit' command properly exits the program
+        mocked_input.side_effect = [EXIT]
+        self.session.start()
+        self.assertRaises(SystemExit)
+
+        # Testing that all commands redirect to their respective handlers in commands.py
+        command_class_path = "src.Session.commands.Commands"
+        self.checkCommandMethodCalled(mocked_input, [ADD_FOOD, EXIT], command_class_path + ".add_food")
+        self.checkCommandMethodCalled(mocked_input, [CREATE_FOOD_STORAGE, EXIT], command_class_path + ".create_foodStorage", FOOD_STORAGE)
+        self.checkCommandMethodCalled(mocked_input, [CREATE_FREEZER, EXIT], command_class_path + ".create_foodStorage", FREEZER)
+        self.checkCommandMethodCalled(mocked_input, [CREATE_FRIDGE, EXIT], command_class_path + ".create_foodStorage", FRIDGE)
+        self.checkCommandMethodCalled(mocked_input, [CREATE_PROFILE, EXIT], command_class_path + ".create_profile")
+        self.checkCommandMethodCalled(mocked_input, [DELETE_FOOD_STORAGE, EXIT], command_class_path + ".delete", FOOD_STORAGE)
+        self.checkCommandMethodCalled(mocked_input, [DELETE_FREEZER, EXIT], command_class_path + ".delete", FREEZER)
+        self.checkCommandMethodCalled(mocked_input, [DELETE_FRIDGE, EXIT], command_class_path + ".delete", FRIDGE)
+        self.checkCommandMethodCalled(mocked_input, [DELETE_PROFILE, EXIT], command_class_path + ".delete_profile")
+        self.checkCommandMethodCalled(mocked_input, [HELP, EXIT], command_class_path + ".help")
+        self.checkCommandMethodCalled(mocked_input, [LIST_FOOD_STORAGES, EXIT], command_class_path + ".list_food_storages", FOOD_STORAGE)
+        self.checkCommandMethodCalled(mocked_input, [LIST_FREEZERS, EXIT], command_class_path + ".list_food_storages", FREEZER)
+        self.checkCommandMethodCalled(mocked_input, [LIST_FRIDGES, EXIT], command_class_path + ".list_food_storages", FRIDGE)
+        self.checkCommandMethodCalled(mocked_input, [LOAD, EXIT], command_class_path + ".load")
+        self.checkCommandMethodCalled(mocked_input, [OPEN_FOOD_STORAGE, EXIT], command_class_path + ".open", FOOD_STORAGE)
+        self.checkCommandMethodCalled(mocked_input, [OPEN_FREEZER, EXIT], command_class_path + ".open", FREEZER)
+        self.checkCommandMethodCalled(mocked_input, [OPEN_FRIDGE, EXIT], command_class_path + ".open", FRIDGE)
+        self.checkCommandMethodCalled(mocked_input, [REMOVE_FOOD, EXIT], command_class_path + ".remove_food")
+        self.checkCommandMethodCalled(mocked_input, [SAVE, EXIT], command_class_path + ".save")
```

Comment:
I like how you did this, very cool.

Reply:
thank you! it actually took me a long time to figure out how to get it to work lol i was having a hard time figuring out how to check that a certain command was called. Let me know if you need help understanding this, I know it's a lot!